docs/book/how-to/customize-docker-builds/how-to-reuse-builds.md

```diff
@@ -1,8 +1,39 @@
+# How to reuse builds
 
-You can do so by connecting a git repository. Registering a code repository lets you avoid building images each time you run a pipeline **and** quickly iterate on your code. When running a pipeline that is part of a local code repository checkout, ZenML can instead build the Docker images without including any of your source files, and download the files inside the container before running your code. This greatly speeds up the building process and also allows you to reuse images that one of your colleagues might have built for the same stack.
+## What is a build?
+
+A pipeline build is an encapsulation of a pipeline and the stack it was run on. It contains the Docker images that were built for the pipeline with all the requirements from the stack, integrations and the user. Optionally, it also contains the pipeline code.
+
+When you run a pipeline, ZenML will check if a build with the same pipeline and stack exists. If it does, it will reuse that build. If it doesn't, ZenML will create a new build.
+
+You can list all the builds for a pipeline using the CLI:
+
+```bash
+zenml pipeline builds list --pipeline_id='startswith:ab53ca'
+```
+
+You can also create a build manually using the CLI:
+
+```bash
+zenml pipeline build --stack vertex-stack my_module.my_pipeline_instance
+```
+
+You can use the options to specify the configuration file and the stack to use for the build. The source should be a path to a pipeline instance. Learn more about the build function [here](https://sdkdocs.zenml.io/latest/core_code_docs/core-new/#zenml.new.pipelines.pipeline.Pipeline.build).
+
+## Reusing builds
+
+As already mentioned, ZenML will find an existing build if it matches your pipeline and stack, by itself. However, you can also force it to use a specific build by [passing the build ID](../../how-to/use-configuration-files/what-can-be-configured.md#build-id) to the `build` parameter of the pipeline configuration.
+
+While reusing Docker builds is useful, it can be limited. This is because specifying a custom build when running a pipeline will **not run the code on your client machine** but will use the code **included in the Docker images of the build**. As a consequence, even if you make local code changes, reusing a build will _always_ execute the code bundled in the Docker image, rather than the local code. Therefore, if you would like to reuse a Docker build AND make sure your local code changes are also downloaded into the image, you need to disconnect your code from the build. You can do this either by registering a code repository or by letting ZenML use the artifact store to upload your code.
```

Comment:
if you reuse a build without having a code repo, it would mean that it included your code in the build's image (provided that the download from artifact flag was also False). the next section tells you to reuse builds but have a code repo and recommends that as best practice because in that case, the code is not included in your build image.

at least that's my understanding but maybe michael can shed some light